src/nile/core/deploy.py

```diff
@@ -22,5 +31,8 @@ def deploy(contract_name, arguments, network, alias, overriding_path=None, abi=N
     )
     logging.info(f"🧾 Transaction hash: {hex(tx_hash)}")
 
+    if status_type is not None:
+        status(tx_hash, network, status_type)
```

Comment:
I think may be a good idea to do this after the full execution of the command (at the end of CLI method, for example) because if the user cancels the execution in the terminal, the accounts and deployments files are not going to be updated, potentially making the user need to re-deploy, or to manually add things (poor UX), even when the transaction was successfully sent.

The suggestion is to block the execution only after the complete execution of the command (this can be better tested in goerli where there is a delay instead of devnet).

Reply:
Good point! My reservation with moving the status check/command outside of the core lies in the fact that we'd need to repeat the same steps in the NRE (assuming that we want this functionality in the NRE)—open to opinions here.

Otherwise, an easy solution could be to place the status call _after_ deployment registration:

```
# core/deploy.py

def deploy(...):
    ...

    deployments.register(...)

    if status_type is not None:
        status(...)

    return address, register_abi
```

Reply:
> Good point! My reservation with moving the status check/command outside of the core lies in the fact that we'd need to repeat the same steps in the NRE (assuming that we want this functionality in the NRE)—open to opinions here.
>

I think it is most valuable to support it in the NRE if users deploy within it.

> Otherwise, an easy solution could be to place the status call _after_ deployment registration:
>

In that case, it shouldn't be too difficult to *un*register on failed transactions either.

Reply:
> I think it is most valuable to support it in the NRE if users deploy within it.

Agreed^

> In that case, it shouldn't be too difficult to unregister on failed transactions either.

Yes, that's a great idea!